### PR TITLE
Remove all usages of MailPoet\Tasks\Sending::getRunningQueues()[MAILPOET-4366]

### PR DESCRIPTION
**Pull request by lysyjan**
[MAILPOET-4366]

[MAILPOET-4366]: https://mailpoet.atlassian.net/browse/MAILPOET-4366?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/Newsletter/Sending/ScheduledTasksRepository.php [hunk 1]
```diff
@@ -125,6 +125,25 @@ public function findFutureScheduledByType($type, $limit = null) {
...
+   * @return ScheduledTaskEntity[]
+   */
+  public function findRunningSendingTasks(?int $limit = null): array {
+    return $this->doctrineRepository->createQueryBuilder('st')
+      ->select('st')
+      ->join('st.sendingQueue', 'sq')
+      ->where('st.type = :type')
+      ->andWhere('st.status IS NULL')
+      ->andWhere('st.deletedAt IS NULL')
+      ->andWhere('sq.deletedAt IS NULL')
```
Comment (costasovo) [inline mailpoet/lib/Newsletter/Sending/ScheduledTasksRepository.php]: I noticed that [the original query](https://github.com/mailpoet/mailpoet/blob/9b71a7804cd7d3c8724c9ac877cc62def4d3a697/mailpoet/lib/Tasks/Sending.php#L317-L326) was not checking `deletedAt` in the sending_queue table.
Do you think it is necessary to check it? I mean, we could avoid the join here and maybe make the query more efficient.

File: mailpoet/lib/Entities/ScheduledTaskEntity.php [hunk 2]
```diff
@@ -21,6 +21,7 @@ class ScheduledTaskEntity {
   const STATUS_PAUSED = 'paused';
   const STATUS_INVALID = 'invalid';
   const VIRTUAL_STATUS_RUNNING = 'running'; // For historical reasons this is stored as null in DB
+  const TYPE_SENDING = 'sending';
```
Comment (costasovo) [inline mailpoet/lib/Entities/ScheduledTaskEntity.php]: Other scheduled task type constants are declared in related worker classes (e.g. [#](https://github.com/mailpoet/mailpoet/blob/5373c811b2525de7a586e850a2c8fa9cfc04c168/mailpoet/lib/Cron/Workers/Bounce.php#L24), [#](https://github.com/mailpoet/mailpoet/blob/af396021db8ee4da756409d53c731b08c93c6926/mailpoet/lib/Cron/Workers/InactiveSubscribers.php#L12)).
So I would suggest declaring this type constant in [the SendingQueue worker](https://github.com/mailpoet/mailpoet/blob/8d4799b050bf058b8bb73ceb056c5f02ca0a33ec/mailpoet/lib/Cron/Workers/SendingQueue/SendingQueue.php#L32).
Also please note that this constant is already defined in https://github.com/mailpoet/mailpoet/blob/9b71a7804cd7d3c8724c9ac877cc62def4d3a697/mailpoet/lib/Tasks/Sending.php#L26
So maybe we could remove it from there.

**Review by costasovo — COMMENTED** [review]
Good job! Overall looks good! I have just one remark and one question.

File: mailpoet/lib/Newsletter/Sending/ScheduledTasksRepository.php (hunk 1 above)
Comment (lysyjan) [reply mailpoet/lib/Newsletter/Sending/ScheduledTasksRepository.php]: You are right. We can remove the condition for sending queues.

File: mailpoet/lib/Entities/ScheduledTaskEntity.php (hunk 2 above)
Comment (lysyjan) [reply mailpoet/lib/Entities/ScheduledTaskEntity.php]: Thank you for this tip. I agree!

**Review by costasovo — DISMISSED** [review]

**Review by veljkho — CHANGES_REQUESTED** [review]
Seems not verified to me
I'm getting [call to undefined method](https://d.pr/i/3SDjUy) error when accessing `Subscribers Engagement` tab in newsletter's statistics page.

**Review by veljkho — APPROVED** [review]

**Comment by veljkho** [comment]
Seems verified after rebasing it. Going to merge.

**Pull request merged**